services/web/server/src/simcore_service_webserver/meta_handlers.py

```diff
@@ -0,0 +1,277 @@
+""" web-api handler functions added by the meta app's module
+
+"""
+import logging
+from typing import List, Optional, Tuple
+
+from aiohttp import web
+from models_library.projects import ProjectID
+from models_library.rest_pagination import DEFAULT_NUMBER_OF_ITEMS_PER_PAGE, Page
+from models_library.rest_pagination_utils import paginate_data
+from pydantic import BaseModel
+from pydantic.fields import Field
+from pydantic.networks import HttpUrl
+
+from ._meta import api_version_prefix as VTAG
+from .meta_iterations import ProjectIteration
+from .meta_version_control import VersionControlForMetaModeling
+from .rest_utils import RESPONSE_MODEL_POLICY
+from .security_decorators import permission_required
+from .utils_aiohttp import create_url_for_function, envelope_json_response
+from .version_control_models import CheckpointID, CommitID, TagProxy
+from .version_control_tags import parse_workcopy_project_tag_name
+
+log = logging.getLogger(__name__)
+
+
+# HANDLER'S CORE IMPLEMENTATION ------------------------------------------------------------
+
+IterationTuple = Tuple[ProjectID, CommitID]
+
+
+async def list_project_iterations(
+    vc_repo: VersionControlForMetaModeling,
+    project_uuid: ProjectID,
+    commit_id: CommitID,
+    offset: int = 0,
+    limit: Optional[int] = None,
+) -> Tuple[List[IterationTuple], int]:
+
+    assert offset >= 0  # nosec
+
+    repo_id = await vc_repo.get_repo_id(project_uuid)
+    assert repo_id is not None
+
+    total_number_of_iterations = 0  # count number of iterations
+
+    # Search all subsequent commits (i.e. children) and retrieve their tags
+    # Select range on those tagged as iterations and returned their assigned workcopy id
+    # Can also check all associated project uuids and see if they exists
+    # FIXME: do all these operations in database
```

Comment:
is this a FIXME or a TODO??
if it does not work it is a FIXME, if it works but is slow it's a TODO agree? also FIXMEs all over the place are scarry

Reply:
I cleaned up